projectile.el

```diff
@@ -3182,25 +3182,38 @@ Should be set via .dir-locals.el.")
 It takes precedence over the default command for the project type when set.
 Should be set via .dir-locals.el.")
 
+(defun projectile-default-generic-command (project-type command-type)
+  "Generic retrieval of COMMAND-TYPEs default cmd-value for PROJECT-TYPE.
+
+If found, checks if value is symbol or string. In case of symbol resolves
+to function `funcall's. Return value of function MUST be string to be executed as command."
+  (let ((plist-retrieved-generic-command (plist-get (gethash project-type projectile-project-types) command-type)))
+    (cond
```

Comment:
Probably this `cond` should have a default branch raising some error if the command is not a symbol or a string.

Reply:
For the error condition `(cond (t ... ))` what would your preferred method of warning be? Simple `(error "")` or would you prefer a full `(throw ... )`?

Also it'd be interesting to know how to test this. I was thinking about this:
```elisp
  (catch 'err
      (projectile-default-compilation-command 'has-command-at-point))
    (error
     (should
      (equal "The value for: default-compilation-command in project-type: has-command-at-point was neither a function nor a string."
             (car (cdr err))))))
```

But that seems to fail in ert. If there is a better way I'm happy to learn. :)